src/lib.rs

```diff
@@ -179,6 +179,23 @@ where
     /// * `ey` - y coordinate end
     /// * `colors` - anything that can provide `IntoIterator<Item = u16>` to iterate over pixel data
     ///
+    /// # Example
```

Comment:
If you want to improve the docs for `set_pixel` further:

One thing that is important to document IMO is the behavior of `set_pixels` if the `colors` iterator contains more pixels than required by the rectangle bounds. If I remember correctly there is no bounds checking and the drawing operation would just wrap back to the top left of the rectangle.

This is different to `embedded-graphics`'s similar [`fill_contiguous` method](https://docs.rs/embedded-graphics/latest/embedded_graphics/draw_target/trait.DrawTarget.html#method.fill_contiguous) and should be highlighted. Because this makes it impossible to just use something like `display.set_pixel(0, 0, 10, 10, core::iter::repeat(Rgb565::RED))`, which works in e-g.

Reply:
> This is different to embedded-graphics's similar [`fill_contiguous` method](https://docs.rs/embedded-graphics/latest/embedded_graphics/draw_target/trait.DrawTarget.html#method.fill_contiguous) and should be highlighted

I added a `# Warning` [**here »**](https://github.com/almindor/mipidsi/pull/66/commits/e611757fa4155a66efd5724e5a1d50896a0f739f#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R192) mentioning that using [`fill_contiguous`](https://docs.rs/embedded-graphics/latest/embedded_graphics/draw_target/trait.DrawTarget.html#method.fill_contiguous) instead is recommended. Does this sound good or should I highlight that in another way?

Reply:
I would move it to above the argument list and rephrase it a bit. It's quite important to know the behavior of and IMO it's harder to notice the warning if it is below the example.

Instead of a warning I would just describe the behavior of this method. A rough draft of a description:
> No bounds checking is performed on the `colors` iterator, which can lead to unexpected behavior if the iterator returns more or less color values than the number of pixels in the given update region. Use [`fill_contiguous`](https://docs.rs/embedded-graphics/latest/embedded_graphics/draw_target/trait.DrawTarget.html#method.fill_contiguous) from `embedded-graphics` as an alternative with bounds checking.

Reply:
One note here is that `set_pixels` is essentially a driver primitive. It's not meant to be used directly and is exposed so frameworks other than `embedded-graphics` can be used.